src/cli.jl

```diff
@@ -59,80 +72,108 @@ treeknit
 
 	global_logger(TeeLogger(loggers...))
 
-
 	# Reading trees
-	@info "Input Newick files: $nwk1 \t $nwk2. Reading trees..."
-	t1 = read_tree(nwk1)
-	t2 = read_tree(nwk2)
-	if !TreeTools.share_labels(t1, t2)
-		error("Trees must share leaves")
+	@info "Input Newick files:"*join(["$nwk \t" for nwk in nwk_files])*". Reading trees..."
+	fn, ext = get_tree_names(nwk_files)
+	trees = [read_tree(nwk, label=f) for (nwk, f) in zip(nwk_files, fn)]
+	for t_i in trees[2:end] 
+		if !TreeTools.share_labels(trees[1], t_i)
+			error("Trees must share leaves")
+		end
 	end
 
 	# Reading sequence lengths
 	sl = try
 		local sl = map(i->parse(Int, i), split(seq_lengths, " "))
-		@assert length(sl) == 2
+		@assert length(sl) == length(trees)
 		sl
 	catch err
 		@error "Unrecognized format for `--seq-lengths`.
 Should be of the form `--seq-lengths \"1500 2000\"`"
 		error(err)
 	end
 
+	##only parallelize if there are 4 or more trees
+	if length(trees)<=3 && parallel
+		@warn "Not running in parallel for less than 4 trees (got $(length(trees)))"
+		parallel = false
+	end
+
+	println("Performing $rounds rounds of TreeKnit")
+	if resolve_all_rounds || (length(trees)==2 && !final_no_resolve)
```

Comment:
Not clear on first read which resolve option overrides which. I'll give it a closer look, would be useful to have this explicitly in the docstring.